#991 Support Relative Date Times (#1006)
opensearch-project/opensearch-spark: docs/ppl-lang/functions/ppl-datetime.md

    | 3                             |
    +-------------------------------+

### `RELATIVE_TIMESTAMP`

**Description:**

**Usage:** relative_timestamp(str) returns a timestamp corresponding to the give relative string and the current
timestamp at the time of query execution.

The relative time string has syntax `[+|-]<offset_time_integer><offset_time_unit>@<snap_time_unit>`, and is made up of
two optional components:
* An offset from the current timestamp at the start of query execution, which is composed of a sign (`+` or `-`), an 

>   optional time integer, and a time unit. If the time integer is not specified, it defaults to one. For example, `+2hr`

Reviewer: ```suggestion
 `offset_time_integer` integer, and `offset_time_unit` a time unit. If the time integer is not specified, it defaults to`1`. For example, `+2hr`
```

Author: Thanks, done ✅

  corresponds to two hours after the current timestamp, while `-mon` corresponds to one month ago. 

> * A snap-to time using the `@` symbol followed by a time unit. The snap-to time is applied after the offset (if 

Reviewer: ```suggestion
* A snap-to time using the `@` symbol followed by `snap_time_unit` - a time unit. The snap-to time is applied after the offset (if 
```

Author: Thanks, done ✅

>   specified), and rounds the time <i>down</i> to the start of the specified time unit (i.e. backwards in time). For 

Reviewer: ```suggestion
  specified), and rounds the time <i>down</i> to the start of the specified time unit. For 
```

Author: Done ✅

  example, `@wk` corresponds to the start of the current week (Sunday is considered to be the first day of the week).

> The following offset time units are supported:

Reviewer: case-sensitive?

Author: Good call - forgot to mention that! I have added a line a bit further down that "the entire
relative timestamp string is case-insensitive", and also updated two of the example to use capital letters to illustrate this

| Time Unit | Supported Keywords                        |
|-----------|-------------------------------------------|
| Seconds   | `s`, `sec`, `secs`, `second`, `seconds`   |
| Minutes   | `m`, `min`, `mins`, `minute`, `minutes`   |
| Hours     | `h`, `hr`, `hrs`, `hour`, `hours`         |
| Days      | `d`, `day`, `days`                        |
| Weeks     | `w`, `wk`, `wks`, `week`, `weeks`         |
| Quarters  | `q`, `qtr`, `qtrs`, `quarter`, `quarters` |
| Years     | `y`, `yr`, `yrs`, `year`, `years`         |

> The snap-to time supports all the time units above, as well as the following day of the week time units:

Reviewer: ```suggestion
The snap-to time supports all the time units above, as well as the following day-of-the-week time units:
```

Author: Done ✅

| Time Unit | Supported Keywords |
|-----------|--------------------|
| Sunday    | `w0`, `w7`         |
| Monday    | `w1`               |
| Tuesday   | `w2`               |
| Wednesday | `w3`               |
| Thursday  | `w4`               |
| Friday    | `w5`               |
| Saturday  | `w6`               |

> The special relative time string `now` for the current timestamp is also supported.

Reviewer: this needs to be mentioned higher up (like on the first line of the usage/description).

Reviewer: Mention that `now` is calculated once at the start of the query execution, and used for relative datetime calculations for that query.

Reviewer: wait... can you verify the above comment?  If you print out `now()` twice, does it give exactly the same result (down to the millisecond)?

Author: Moved the mention of `now` closer to the start, and specified that the current timestamp is the same for all relative timestamps in the same query. Let me know what you think.
